### PR TITLE
Late night float fun

### PR DESCRIPTION
**Pull request by pimeys**
Fixes all kinds of small conversion errors on PostgreSQL numeric, real and float types.

### CODE REVIEW

File: src/connector/postgres/conversion.rs [hunk 1]
```diff
@@ -499,12 +499,29 @@ impl<'a> ToSql for Value<'a> {
...
+            (Value::Integer(integer), &PostgresType::NUMERIC) => integer
+                .map(|integer| BigDecimal::from_i64(integer).unwrap())
+                .map(|bd| DecimalWrapper(bd))
+                .map(|dw| dw.to_sql(ty, out)),
             (Value::Integer(integer), &PostgresType::TEXT) => {
                 integer.map(|integer| format!("{}", integer).to_sql(ty, out))
             }
             (Value::Integer(integer), &PostgresType::OID) => integer.map(|integer| (integer as u32).to_sql(ty, out)),
             (Value::Integer(integer), _) => integer.map(|integer| (integer as i64).to_sql(ty, out)),
+            (Value::Float(double), &PostgresType::FLOAT8) => double.map(|double| (double as f64).to_sql(ty, out)),
```
Comment (tomhoule) [inline src/connector/postgres/conversion.rs:512]: nitpick: is `double` the right variable name here?

Also the f32 -> f64 casts behave weirdly, but I guess there isn't a straightforward way around it.

File: src/connector/postgres/conversion.rs [hunk 2]
```diff
@@ -499,12 +499,29 @@ impl<'a> ToSql for Value<'a> {
         let res = match (self, ty) {
             (Value::Integer(integer), &PostgresType::INT2) => integer.map(|integer| (integer as i16).to_sql(ty, out)),
             (Value::Integer(integer), &PostgresType::INT4) => integer.map(|integer| (integer as i32).to_sql(ty, out)),
+            #[cfg(feature = "bigdecimal")]
+            (Value::Integer(integer), &PostgresType::NUMERIC) => integer
+                .map(|integer| BigDecimal::from_i64(integer).unwrap())
+                .map(|bd| DecimalWrapper(bd))
+                .map(|dw| dw.to_sql(ty, out)),
```
Comment (tomhoule) [inline src/connector/postgres/conversion.rs:506]: :+1:

File: src/connector/postgres/conversion.rs [hunk 3]
```diff
@@ -499,12 +499,29 @@ impl<'a> ToSql for Value<'a> {
...
                 integer.map(|integer| format!("{}", integer).to_sql(ty, out))
             }
             (Value::Integer(integer), &PostgresType::OID) => integer.map(|integer| (integer as u32).to_sql(ty, out)),
             (Value::Integer(integer), _) => integer.map(|integer| (integer as i64).to_sql(ty, out)),
+            (Value::Float(double), &PostgresType::FLOAT8) => double.map(|double| (double as f64).to_sql(ty, out)),
+            #[cfg(feature = "bigdecimal")]
+            (Value::Float(float), &PostgresType::NUMERIC) => float
+                .map(|float| BigDecimal::from_f32(float).unwrap())
+                .map(DecimalWrapper)
+                .map(|dw| dw.to_sql(ty, out)),
```
Comment (tomhoule) [inline src/connector/postgres/conversion.rs:517]: :+1:

**Review by tomhoule — APPROVED** [review]

File: src/connector/postgres/conversion.rs (hunk 1 above)
Comment (pimeys) [reply src/connector/postgres/conversion.rs:512]: It's so much fun to write these that I probably spent an hour just focusing on getting the best variable name here last night...

**Pull request merged**